llmc/matmul.cuh

```diff
@@ -14,70 +14,64 @@ Matrix Multiplication, with help from cuBLASLt
 template<typename OutFloat, bool UseAuxBuffer>
 __global__ void matmul_backward_bias_kernel9(OutFloat* dbias, const floatX* dout, int B, int T, int OC,
                                              std::bool_constant<UseAuxBuffer>) {
-    constexpr const int bdx = 4;
-    constexpr const int bdy = WARP_SIZE / bdx;
-    assert(blockDim.x == bdx);
-    assert(blockDim.y == bdy);
 
-    int warp_d = (int)threadIdx.x;
-    int warp_c = (int)threadIdx.y;
-    int block_d = (int)threadIdx.z;
-
-    const int OC_per_warp = bdy * x128::size;  // 64 at BF16
-
-    int local_oc = warp_c * x128::size;
-    int global_oc = blockIdx.x * OC_per_warp + local_oc;
-
-    int local_bt = warp_d + bdx * block_d;
-    int bt_per_block = bdx * blockDim.z;
-
-    float accumulators[x128::size];
-    for (int k = 0; k < x128::size; k++) {
-        accumulators[k] = 0.0f;
-    }
-
-    if(global_oc < OC) {
-        // sum up over all bt within registers
-        for (int idx = blockIdx.y * bt_per_block + local_bt; idx < B * T; idx += gridDim.y * bt_per_block) {
-            x128 packed_dout = load128(dout + global_oc + idx*OC);
+        static_assert(WARP_SIZE % 4 == 0, "WARP_SIZE must be a multiple of 4");
+        constexpr const int bdx = 4;
+        constexpr const int bdy = WARP_SIZE / bdx;
+        assert(blockDim.x == bdx && blockDim.y == bdy);
+    
+        int warp_d = threadIdx.x, warp_c = threadIdx.y, block_d = threadIdx.z;
+        int global_oc = blockIdx.x * bdy * x128::size + warp_c * x128::size; // 64 OCs at BF16
+    
+        if (global_oc < OC) {
+            int bt_per_block = bdx * blockDim.z;
+            float accumulators[x128::size] = {0};
```

Comment:
this is horrible syntax. The unsuspecting reader might see this and apply it to, e.g., 
`float accumulators[8] = {1};`, expecting to get an array of ones, and being disappointed